core/src/com/unciv/logic/automation/unit/UnitAutomation.kt

```diff
@@ -181,7 +181,9 @@ object UnitAutomation {
             // Restrict Human automated units from promotions via setting
                 (UncivGame.Current.settings.automatedUnitsChoosePromotions || unit.civ.isAI())) {
             val availablePromotions = unit.promotions.getAvailablePromotions()
-                .filterNot { it.hasUnique(UniqueType.SkipPromotion) }
+            if (unit.health < 60 && !(unit.baseUnit.isAirUnit() || unit.baseUnit.hasUnique(UniqueType.CanMoveAfterAttacking)))
+                availablePromotions.filter { it.hasUnique(UniqueType.OneTimeUnitHeal) } //choose healing promotions only when beneficial
+            else availablePromotions.filterNot { it.hasUnique(UniqueType.SkipPromotion) }
```

Comment:
What if we don't have any promotion with a one time heal? We should default to all

Reply:
I suppose this works.